contracts/wXTMBridge.sol

```diff
@@ -1,30 +1,70 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.22;
 
-import {EIP3009} from "./extensions/EIP3009.sol";
-import {ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import { EIP3009 } from "./extensions/EIP3009.sol";
+/** @dev Ownable from openzeppelin conflict -> to be resolved */
+// import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
 
-import {IwXTM} from "./interfaces/IwXTM.sol";
+import { IwXTM } from "./interfaces/IwXTM.sol";
 
 contract wXTMBridge is EIP3009 {
-    address private immutable wXTM;
+    error wXTMBridge_NotOwner();
+
+    address public owner;
+    address private wXTM;
 
     event TokensMinted(address indexed to, uint256 indexed amount, bytes32 indexed authorizationId);
     event TokensBurned(address indexed from, uint256 indexed amount, bytes32 indexed authorizationId);
 
-    constructor(address _wXTM, string memory version) EIP3009("wXTMBridge", version) {
+    constructor(address _wXTM, address _delegate) {
         wXTM = _wXTM;
+        owner = _delegate;
+    }
+
+    /** @dev Function to handle wXTM upgradable concept */
+    function updatewXTMAddress(address wXTMAddress) external onlyOwner {
```

Comment:
this doesn't need to exist
wXTM will not change address
Here you will have an address of delegation proxy which remains constant during upgrades
safe to remove this function